### ISSUE TITLE
Feature request: paste source name and URL

### ISSUE THREAD

**Issue by graphit0** [sanzoghenzo/markdownr#5]
As a quality of life enhancement, option of pasting the URL itself and possibly the page name in the note would be a neat addition.

This way you could easily search and organize notes by the source in the future.

Thank you for your contribution to open source! Your project greatly improved mobile experience of my Joplin note taking.

**Comment by sanzoghenzo** [comment]
Hi @graphit0 ,
I'm glad this is useful to you!
And thanks for this feature request, it would be a great addition indeed!

How do you imagine this can be added? Two simple checkboxes in the main screen (to easily toggle them on a per page basis) or a global configuration in a separate view?

Also, how should they be inserted in the markdown? I'm inclined to use the yaml front matter to store these kind of thing, for example:

```
---
title: my web page
source: https://example.com/page.html
---

[Page markdown]
```

This works well with dendron (my current note taking app), but I can see that joplin doesn’t play nice with it.

Could you give me an example of how you normally/manually insert these info?

**Comment by graphit0** [comment]
Thank you for the lightning fast response,

>Also, how should they be inserted in the markdown? I'm inclined to use the yaml front matter to store these kind of thing

I'm absolutely in love with yaml front matter but unfortunately, in my experience it faces user friendliness problems. Non-tech people don't understand it and often get confused. Moreover providing editable and a tad fragile tags to user may result in bad experience due to user error.

Just now I tried inputting front matter tags to [GitJournal](https://github.com/GitJournal/GitJournal) (who claims to support it) via share intent, it created a duplicate dashblock (`--- tag:X ---`) and the editor refused to recognise its content. Only after editing manually, the editor accepted the provided/shared tags. This experience is especially undesirable on mobile with screen keyboard. However it pains me, this is the state of front matter adoption right now: those bold few apps who support it may suffer from bad user experience 😮‍💨

Btw, Joplin does support (i.e. can read) front matter but its editor doesn't do much with it. As many other note taking apps, unless instructed otherwise, Joplin fails to ignore it and tries to shove front matter tags into a title.

> How do you imagine this can be added? Two simple checkboxes in the main screen (to easily toggle them on a per page basis) or a global configuration in a separate view?

At the current stage the app doesn't have/need settings but maybe someday it might. That would probably entail a separate settings screen. But for now, to avoid overcomplication, I think better to start with a simple UI element. 

### My take on this element
`kebab` menu dropdown at the top right with a checkbox "Include link 🔗". Its state should be kept persistently (not being reset after each restart)
![kebab menu](https://user-images.githubusercontent.com/44114323/171141632-d2dc60ec-95d6-4c1f-81ca-15146807cdcb.png)

In terms of webpage `title`, I believe it should be included by default hence doesn't need the checkbox. But I'm not adamant about this.

Both of those checkboxes insert generic, markdown formatted, text lines

```
# Title My Web Page
[blank line] // to force markdown standard to register next text line as a separate one
Clipped from: https://example.com/page.html
```

### Front matterize
Although it is out of scope of this FR, to achieve some sort of compromise about front matter, I think a second checkbox could be added: "Front matterize". This feature would replace title, url and input any other tag (in case you'd like to track them) into front matter format: such as location, device name, etc.

Sorry for weird naming, I don't know how to put "front matter" into verb form and manage to cram it into a single line.

**Comment by sanzoghenzo** [comment]
Thank you for the great explanation!

I totally agree with everything.
Front matter can wait, but the option you envision looks good.

I can't promise any ETA/quick resolution for this since I'm moving to a new home...

Obviously PRs are very welcome!

**Comment by graphit0** [comment]
Thank you so much for this!
It works great and significantly improved visibility of my clipped content. I'll eagerly recommend your project to markdown note taking community.
